pkg/deviceplugin/api.go

```diff
@@ -66,6 +66,11 @@ func NewDeviceInfo(state string, nodes []pluginapi.DeviceSpec, mounts []pluginap
 	return deviceInfo
 }
 
+// SetTopology sets the topology information in DeviceInfo struct
+func (d *DeviceInfo) SetTopology(topologyInfo *pluginapi.TopologyInfo) {
+	d.topology = topologyInfo
```

Comment:
Unless you're going to call this setter more than once for a `DeviceInfo` I think it would be better to introduce a new constructor `NewDeviceInfoWithTopologyHints()` or smth. This way we wouldn't call `topology.GetTopologyInfo()` needlessly.